keel-orca/src/main/kotlin/com/netflix/spinnaker/keel/orca/OrcaService.kt

```diff
@@ -16,18 +16,30 @@
 package com.netflix.spinnaker.keel.orca
 
 import com.netflix.spinnaker.keel.model.OrchestrationRequest
-import retrofit.http.Body
-import retrofit.http.Headers
-import retrofit.http.POST
+import retrofit.http.*
 
 // TODO Origin needs to be set on executions
+// origin is used for dynamic routing by orca to different clouddriver instances
 interface OrcaService {
 
   @POST("/ops")
   @Headers("Content-Type: application/context+json")
   fun orchestrate(@Body request: OrchestrationRequest): TaskRefResponse
+
+  @GET("/tasks/{id}")
+  fun getTask(@Path("id") id: String): TaskDetailResponse
 }
 
 data class TaskRefResponse(
   val ref: String
 )
+
+data class TaskDetailResponse(
+  val id: String,
+  val name: String,
+  val application: String,
+  val buildTime: String,
+  val startTime: String,
+  val endTime: String,
```

Comment:
Since we get endTime from orca, emitting a intent.orchestrations.duration metric could be useful. Especially if it's calculated from a keel specific start time (i.e. using the score/insertion time of orchestration keys in the IntentActivity zsets) instead of the orca start time.

Reply:
I definitely think this might be useful in the future, but I'd like to add this once we want to measure this. 

Currently only the redis implementation keeps track of score, so I suspect I should do more  work on the in memory implementation to have it match the redis one more closely.

I'll add a todo for myself.